activerecord/lib/active_record/associations/foreign_association.rb

```diff
@@ -22,8 +22,15 @@ def nullified_owner_attributes
       def set_owner_attributes(record)
         return if options[:through]
 
-        key = owner._read_attribute(reflection.join_foreign_key)
-        record._write_attribute(reflection.join_primary_key, key)
+        primary_key = Array.wrap(reflection.join_primary_key)
```

Comment:
```suggestion
        primary_key = Array(reflection.join_primary_key)
```

Reply:
Fixed, thanks!
CI is red but it looks like it's just unstable as rubies `2.7` and `3.1` are green